pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -260,6 +260,15 @@ func (cb *ClusterBuilder) applyDestinationRule(mc *clusterWrapper, clusterMode C
 
 	cb.applyMetadataExchange(opts.mutable.cluster)
```

Comment:
I meant cannot we skip applying metadata exchange filter for external service

Reply:
@hzxuzhonghu this is not the same mx filter I am modifying in xds/filters/filter.go ([here](https://github.com/istio/istio/pull/46579/files#diff-96471a6888b5b1ef2c0553d8108f782dd33ebfbb41fed0daac43dd98599bf66fL180)).  This is a tcp mx filter applied at the cluster level.  The filter that is currently appending the headers is applied at the listener level to the http connection manager.  So we can't skip adding the filter there unless we can guarantee no other services are using that XDS listener

Reply:
Ah, IC, will take a nother look